components/public-api/README.md

```diff
@@ -0,0 +1,101 @@
+# Public-API
+
+# Example Flows
+This section gives examples how clients would use this API.
+
+## Create Workspace
+
+### Not prebuild aware
+```go
+workspaces.CreateWorkspace({
+    idempotency_token: "<random_string>",
```

Comment:
Small nit: We could call this `request_id` in order to use shorter & less complicated words on every call. 💡 (Advantage: More easily understood by all users, especially non-native speakers.)

Reply:
Good point you're raising here. There are several choices we have:
1. `idempotency token` or `key`. This accurately describes what's happening. Is used by [IETF](https://tools.ietf.org/id/draft-idempotency-header-01.html#name-the-idempotency-http-reques) and [Stripe](https://stripe.com/docs/api/idempotent_requests). Is long and fancy, hence maybe harder to understand.
2. `client token` is used by AWS. Downside: doesn't relate to the request.
3. `request_id` is short and not fancy (that's good). But it's not self-explanatory either: users would need to read the docs to understand what this field is for and how to use it.

Reply:
`request_id` is often used to mean a trace id. Imo, idempotency token is the most explicit. Users will consume this from sdks and will therefore also see the doc string on this field. That should provide enough clarity (together with website examples)

Reply:
Let's stick to idempotency_token for the reasons above.